#!/bin/sh -e
##
## 02_alpha_blending
##

. debian/patches/patch-opts

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -u libxine.orig/src/xine-engine/alphablend.c libxine/xine-engine/alphablend.c
--- libxine.orig/src/xine-engine/alphablend.c	2005-09-27 02:05:31.000000000 -0000
+++ libxine/xine-engine/alphablend.c	2006-02-05 11:22:47.000000000 -0000
@@ -2019,7 +2019,7 @@
     dst = dst_y;
 
     for (x = 0; x < src_width;) {
-      int len = (x + rle->len > hili_right) ? hili_right - x : rle->len;
+      int len = (x + rle->len > clip_right) ? clip_right - x : rle->len;
       int clipped = (y < clip_top);
 
       if (len > 0) {
